org.mwc.cmap.legacy/src/MWC/TacticalData/NarrativeEntry.java

```diff
@@ -317,16 +334,26 @@ public final String toString()
 //	}
 
 
+	protected void clearHash()
+	{
+	  _hashCode = null;
+	}
+	
 	@Override
   public int hashCode()
   {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + ((_DTG == null) ? 0 : _DTG.hashCode());
-    result = prime * result + ((_entry == null) ? 0 : _entry.hashCode());
-    result = prime * result + ((_track == null) ? 0 : _track.hashCode());
-    result = prime * result + ((_type == null) ? 0 : _type.hashCode());
-    return result;
+	  if(_hashCode == null)
+	  {
+	    final int prime = 31;
+	    int result = 1;
+	    result = prime * result + ((_DTG == null) ? 0 : _DTG.hashCode());

```

Comment:
so we do not need to cache it _dtg should be simple sting and we delegate  'return _DTG.hashCode()'